### PR TITLE
create dirs for 1.18 release

### PR DESCRIPTION
**Pull request by kbhawkey**
- Generate directories and configuration for 1.18 release
- Update kubectl config to generate toc (remove rolling-update)

### CODE REVIEW

File: gen-apidocs/config/v1_18/config.yaml [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  - "AuditRegistration"
+  - "Authentication"
+  - "Authorization"
+  - "Autoscaling"
+  - "Batch"
+  - "Certificates"
+  - "Coordination"
+  - "Core"
+  - "Discovery"
+  - "Extensions"
```
Comment (tengqm) [inline gen-apidocs/config/v1_18/config.yaml]: Looks to me that this group is dropped in 1.18?

Comment (kbhawkey) [reply gen-apidocs/config/v1_18/config.yaml]: OK. Do you think it is a good idea to add the initial files for the release and then iterate on changes?

Comment (tengqm) [reply gen-apidocs/config/v1_18/config.yaml]: of course!

**Comment by kbhawkey** [comment]
Just tried to generate all refs, testing the state of v1.18.
I grabbed a copy of swagger.json (not official 1.18 version).
The kubectl ref generated with the toc.yaml adjusted.
Attached is the error log from trying to generate the API ref.

```
make api
rm -rf /tmp/tmpi5d4u4dd/src/github.com/kubernetes-sigs/reference-docs/gen-apidocs/build
go run gen-apidocs/main.go --kubernetes-release=1.18 --work-dir=gen-apidocs --munge-groups=false
Current Version: storage.v1.VolumeNodeResources Old Versions: [storage.v1beta1.VolumeNodeResources,]
Current Version: apiextensions.v1.CustomResourceDefinitionList Old Versions: [apiextensions.v1beta1.CustomResourceDefinitionList,]
Current Version: autoscaling.v2beta2.MetricSpec Old Versions: [autoscaling.v2beta1.MetricSpec,]
Current Version: rbac.v1.ClusterRoleList Old Versions: [rbac.v1beta1.ClusterRoleList,rbac.v1alpha1.ClusterRoleList,]
Current Version: storage.v1.CSIDriverSpec Old Versions: [storage.v1beta1.CSIDriverSpec,]
Current Version: storage.v1.CSIDriver Old Versions: [storage.v1beta1.CSIDriver,]
Current Version: storage.v1.CSIDriverList Old Versions: [storage.v1beta1.CSIDriverList,]
Current Version: core.v1.EventList Old Versions: [events.v1beta1.EventList,]
Current Version: node.v1beta1.RuntimeClass Old Versions: [node.v1alpha1.RuntimeClass,]
Current Version: apiregistration.v1.APIServiceSpec Old Versions: [apiregistration.v1beta1.APIServiceSpec,]
Current Version: apiextensions.v1.CustomResourceDefinitionVersion Old Versions: [apiextensions.v1beta1.CustomResourceDefinitionVersion,]
Current Version: autoscaling.v1.CrossVersionObjectReference Old Versions: [autoscaling.v2beta2.CrossVersionObjectReference,autoscaling.v2beta1.CrossVersionObjectReference,]
Current Version: authorization.v1.NonResourceAttributes Old Versions: [authorization.v1beta1.NonResourceAttributes,]
Current Version: rbac.v1.Role Old Versions: [rbac.v1beta1.Role,rbac.v1alpha1.Role,]
Current Version: storage.v1.CSINodeList Old Versions: [storage.v1beta1.CSINodeList,]
Current Version: apiextensions.v1.JSONSchemaPropsOrArray Old Versions: [apiextensions.v1beta1.JSONSchemaPropsOrArray,]
Current Version: authorization.v1.SelfSubjectAccessReview Old Versions: [authorization.v1beta1.SelfSubjectAccessReview,]
Current Version: rbac.v1.AggregationRule Old Versions: [rbac.v1beta1.AggregationRule,rbac.v1alpha1.AggregationRule,]
Current Version: authorization.v1.SubjectAccessReview Old Versions: [authorization.v1beta1.SubjectAccessReview,]
Current Version: apiextensions.v1.JSONSchemaProps Old Versions: [apiextensions.v1beta1.JSONSchemaProps,]
Current Version: autoscaling.v2beta2.ResourceMetricSource Old Versions: [autoscaling.v2beta1.ResourceMetricSource,]
Current Version: rbac.v1.ClusterRoleBinding Old Versions: [rbac.v1beta1.ClusterRoleBinding,rbac.v1alpha1.ClusterRoleBinding,]
Current Version: admissionregistration.v1.MutatingWebhook Old Versions: [admissionregistration.v1beta1.MutatingWebhook,]
Current Version: apiextensions.v1.CustomResourceSubresourceStatus Old Versions: [apiextensions.v1beta1.CustomResourceSubresourceStatus,]
Current Version: rbac.v1.RoleList Old Versions: [rbac.v1beta1.RoleList,rbac.v1alpha1.RoleList,]
Current Version: apiextensions.v1.CustomResourceConversion Old Versions: [apiextensions.v1beta1.CustomResourceConversion,]
Current Version: networking.v1beta1.IngressTLS Old Versions: [extensions.v1beta1.IngressTLS,]
Current Version: batch.v1beta1.CronJobSpec Old Versions: [batch.v2alpha1.CronJobSpec,]
Current Version: networking.v1beta1.IngressSpec Old Versions: [extensions.v1beta1.IngressSpec,]
Current Version: rbac.v1.RoleBinding Old Versions: [rbac.v1beta1.RoleBinding,rbac.v1alpha1.RoleBinding,]
Current Version: authorization.v1.NonResourceRule Old Versions: [authorization.v1beta1.NonResourceRule,]
Current Version: authentication.v1.UserInfo Old Versions: [authentication.v1beta1.UserInfo,]
Current Version: networking.v1beta1.IngressList Old Versions: [extensions.v1beta1.IngressList,]
Current Version: authorization.v1.ResourceRule Old Versions: [authorization.v1beta1.ResourceRule,]
Current Version: autoscaling.v1.HorizontalPodAutoscalerStatus Old Versions: [autoscaling.v2beta2.HorizontalPodAutoscalerStatus,autoscaling.v2beta1.HorizontalPodAutoscalerStatus,]
Current Version: autoscaling.v2beta2.ExternalMetricSource Old Versions: [autoscaling.v2beta1.ExternalMetricSource,]
Current Version: authorization.v1.SubjectRulesReviewStatus Old Versions: [authorization.v1beta1.SubjectRulesReviewStatus,]
Current Version: node.v1beta1.Scheduling Old Versions: [node.v1alpha1.Scheduling,]
Current Version: autoscaling.v1.HorizontalPodAutoscalerList Old Versions: [autoscaling.v2beta2.HorizontalPodAutoscalerList,autoscaling.v2beta1.HorizontalPodAutoscalerList,]
Current Version: apiextensions.v1.CustomResourceDefinitionSpec Old Versions: [apiextensions.v1beta1.CustomResourceDefinitionSpec,]
Current Version: authorization.v1.SelfSubjectRulesReview Old Versions: [authorization.v1beta1.SelfSubjectRulesReview,]
Current Version: admissionregistration.v1.ServiceReference Old Versions: [admissionregistration.v1beta1.ServiceReference,apiextensions.v1.ServiceReference,apiextensions.v1beta1.ServiceReference,apiregistration.v1.ServiceReference,apiregistration.v1beta1.ServiceReference,auditregistration.v1alpha1.ServiceReference,]
Current Version: apiregistration.v1.APIService Old Versions: [apiregistration.v1beta1.APIService,]
Current Version: autoscaling.v2beta2.ResourceMetricStatus Old Versions: [autoscaling.v2beta1.ResourceMetricStatus,]
Current Version: storage.v1.CSINodeDriver Old Versions: [storage.v1beta1.CSINodeDriver,]
Current Version: apiextensions.v1.CustomResourceDefinitionCondition Old Versions: [apiextensions.v1beta1.CustomResourceDefinitionCondition,]
Current Version: admissionregistration.v1.MutatingWebhookConfigurationList Old Versions: [admissionregistration.v1beta1.MutatingWebhookConfigurationList,]
Current Version: batch.v1beta1.CronJobStatus Old Versions: [batch.v2alpha1.CronJobStatus,]
Current Version: batch.v1beta1.CronJob Old Versions: [batch.v2alpha1.CronJob,]
Current Version: rbac.v1.ClusterRoleBindingList Old Versions: [rbac.v1beta1.ClusterRoleBindingList,rbac.v1alpha1.ClusterRoleBindingList,]
Current Version: coordination.v1.LeaseList Old Versions: [coordination.v1beta1.LeaseList,]
Current Version: authorization.v1.SelfSubjectRulesReviewSpec Old Versions: [authorization.v1beta1.SelfSubjectRulesReviewSpec,]
Current Version: storage.v1.VolumeAttachmentList Old Versions: [storage.v1beta1.VolumeAttachmentList,storage.v1alpha1.VolumeAttachmentList,]
Current Version: apiextensions.v1.JSONSchemaPropsOrBool Old Versions: [apiextensions.v1beta1.JSONSchemaPropsOrBool,]
Current Version: autoscaling.v2beta2.ObjectMetricSource Old Versions: [autoscaling.v2beta1.ObjectMetricSource,]
Current Version: batch.v1beta1.JobTemplateSpec Old Versions: [batch.v2alpha1.JobTemplateSpec,]
Current Version: autoscaling.v2beta2.HorizontalPodAutoscalerCondition Old Versions: [autoscaling.v2beta1.HorizontalPodAutoscalerCondition,]
Current Version: storage.v1.VolumeAttachmentSource Old Versions: [storage.v1beta1.VolumeAttachmentSource,storage.v1alpha1.VolumeAttachmentSource,]
Current Version: authentication.v1.TokenReview Old Versions: [authentication.v1beta1.TokenReview,]
Current Version: admissionregistration.v1.ValidatingWebhookConfiguration Old Versions: [admissionregistration.v1beta1.ValidatingWebhookConfiguration,]
Current Version: networking.v1beta1.IngressRule Old Versions: [extensions.v1beta1.IngressRule,]
Current Version: networking.v1beta1.IngressStatus Old Versions: [extensions.v1beta1.IngressStatus,]
Current Version: apiextensions.v1.CustomResourceDefinitionStatus Old Versions: [apiextensions.v1beta1.CustomResourceDefinitionStatus,]
Current Version: authentication.v1.TokenReviewStatus Old Versions: [authentication.v1beta1.TokenReviewStatus,]
Current Version: rbac.v1.RoleBindingList Old Versions: [rbac.v1beta1.RoleBindingList,rbac.v1alpha1.RoleBindingList,]
Current Version: storage.v1.CSINode Old Versions: [storage.v1beta1.CSINode,]
Current Version: core.v1.EventSeries Old Versions: [events.v1beta1.EventSeries,]
Current Version: apiextensions.v1.JSON Old Versions: [apiextensions.v1beta1.JSON,]
Current Version: storage.v1.VolumeAttachmentStatus Old Versions: [storage.v1beta1.VolumeAttachmentStatus,storage.v1alpha1.VolumeAttachmentStatus,]
Current Version: coordination.v1.Lease Old Versions: [coordination.v1beta1.Lease,]
Current Version: networking.v1beta1.HTTPIngressRuleValue Old Versions: [extensions.v1beta1.HTTPIngressRuleValue,]
Current Version: authorization.v1.SubjectAccessReviewStatus Old Versions: [authorization.v1beta1.SubjectAccessReviewStatus,]
Current Version: storage.v1.StorageClass Old Versions: [storage.v1beta1.StorageClass,]
Current Version: apiextensions.v1.CustomResourceDefinitionNames Old Versions: [apiextensions.v1beta1.CustomResourceDefinitionNames,]
Current Version: admissionregistration.v1.RuleWithOperations Old Versions: [admissionregistration.v1beta1.RuleWithOperations,]
Current Version: storage.v1.VolumeError Old Versions: [storage.v1beta1.VolumeError,storage.v1alpha1.VolumeError,]
Current Version: autoscaling.v2beta2.MetricStatus Old Versions: [autoscaling.v2beta1.MetricStatus,]
Current Version: authentication.v1.TokenReviewSpec Old Versions: [authentication.v1beta1.TokenReviewSpec,]
Current Version: core.v1.Event Old Versions: [events.v1beta1.Event,]
Current Version: authorization.v1.ResourceAttributes Old Versions: [authorization.v1beta1.ResourceAttributes,]
Current Version: apiregistration.v1.APIServiceStatus Old Versions: [apiregistration.v1beta1.APIServiceStatus,]
Current Version: coordination.v1.LeaseSpec Old Versions: [coordination.v1beta1.LeaseSpec,]
Current Version: flowcontrol.v1alpha1.Subject Old Versions: [rbac.v1.Subject,rbac.v1beta1.Subject,rbac.v1alpha1.Subject,]
Current Version: autoscaling.v1.HorizontalPodAutoscaler Old Versions: [autoscaling.v2beta2.HorizontalPodAutoscaler,autoscaling.v2beta1.HorizontalPodAutoscaler,]
Current Version: networking.v1beta1.IngressBackend Old Versions: [extensions.v1beta1.IngressBackend,]
Current Version: apiregistration.v1.APIServiceList Old Versions: [apiregistration.v1beta1.APIServiceList,]
Current Version: storage.v1.StorageClassList Old Versions: [storage.v1beta1.StorageClassList,]
Current Version: networking.v1beta1.HTTPIngressPath Old Versions: [extensions.v1beta1.HTTPIngressPath,]
Current Version: authorization.v1.SubjectAccessReviewSpec Old Versions: [authorization.v1beta1.SubjectAccessReviewSpec,]
Current Version: autoscaling.v2beta2.PodsMetricStatus Old Versions: [autoscaling.v2beta1.PodsMetricStatus,]
Current Version: apiextensions.v1.CustomResourceValidation Old Versions: [apiextensions.v1beta1.CustomResourceValidation,]
Current Version: scheduling.v1.PriorityClassList Old Versions: [scheduling.v1beta1.PriorityClassList,scheduling.v1alpha1.PriorityClassList,]
Current Version: apiextensions.v1.CustomResourceDefinition Old Versions: [apiextensions.v1beta1.CustomResourceDefinition,]
Current Version: autoscaling.v2beta2.PodsMetricSource Old Versions: [autoscaling.v2beta1.PodsMetricSource,]
Current Version: storage.v1.CSINodeSpec Old Versions: [storage.v1beta1.CSINodeSpec,]
Current Version: autoscaling.v2beta2.ObjectMetricStatus Old Versions: [autoscaling.v2beta1.ObjectMetricStatus,]
Current Version: storage.v1.VolumeAttachmentSpec Old Versions: [storage.v1beta1.VolumeAttachmentSpec,storage.v1alpha1.VolumeAttachmentSpec,]
Current Version: apiextensions.v1.ExternalDocumentation Old Versions: [apiextensions.v1beta1.ExternalDocumentation,]
Current Version: autoscaling.v2beta2.ExternalMetricStatus Old Versions: [autoscaling.v2beta1.ExternalMetricStatus,]
Current Version: storage.v1.VolumeAttachment Old Versions: [storage.v1beta1.VolumeAttachment,storage.v1alpha1.VolumeAttachment,]
Current Version: node.v1beta1.Overhead Old Versions: [node.v1alpha1.Overhead,]
Current Version: apiextensions.v1.CustomResourceColumnDefinition Old Versions: [apiextensions.v1beta1.CustomResourceColumnDefinition,]
Current Version: rbac.v1.PolicyRule Old Versions: [rbac.v1beta1.PolicyRule,rbac.v1alpha1.PolicyRule,]
Current Version: batch.v1beta1.CronJobList Old Versions: [batch.v2alpha1.CronJobList,]
Current Version: authorization.v1.SelfSubjectAccessReviewSpec Old Versions: [authorization.v1beta1.SelfSubjectAccessReviewSpec,]
Current Version: apiextensions.v1.CustomResourceSubresourceScale Old Versions: [apiextensions.v1beta1.CustomResourceSubresourceScale,]
Current Version: rbac.v1.ClusterRole Old Versions: [rbac.v1beta1.ClusterRole,rbac.v1alpha1.ClusterRole,]
Current Version: admissionregistration.v1.MutatingWebhookConfiguration Old Versions: [admissionregistration.v1beta1.MutatingWebhookConfiguration,]
Current Version: admissionregistration.v1.WebhookClientConfig Old Versions: [admissionregistration.v1beta1.WebhookClientConfig,apiextensions.v1.WebhookClientConfig,apiextensions.v1beta1.WebhookClientConfig,auditregistration.v1alpha1.WebhookClientConfig,]
Current Version: autoscaling.v1.HorizontalPodAutoscalerSpec Old Versions: [autoscaling.v2beta2.HorizontalPodAutoscalerSpec,autoscaling.v2beta1.HorizontalPodAutoscalerSpec,]
Current Version: authorization.v1.LocalSubjectAccessReview Old Versions: [authorization.v1beta1.LocalSubjectAccessReview,]
Current Version: rbac.v1.RoleRef Old Versions: [rbac.v1beta1.RoleRef,rbac.v1alpha1.RoleRef,]
Current Version: networking.v1beta1.Ingress Old Versions: [extensions.v1beta1.Ingress,]
Current Version: node.v1beta1.RuntimeClassList Old Versions: [node.v1alpha1.RuntimeClassList,]
Current Version: admissionregistration.v1.ValidatingWebhookConfigurationList Old Versions: [admissionregistration.v1beta1.ValidatingWebhookConfigurationList,]
Current Version: admissionregistration.v1.ValidatingWebhook Old Versions: [admissionregistration.v1beta1.ValidatingWebhook,]
Current Version: apiregistration.v1.APIServiceCondition Old Versions: [apiregistration.v1beta1.APIServiceCondition,]
Current Version: core.v1.EndpointPort Old Versions: [discovery.v1beta1.EndpointPort,]
Current Version: scheduling.v1.PriorityClass Old Versions: [scheduling.v1beta1.PriorityClass,scheduling.v1alpha1.PriorityClass,]
Current Version: apiextensions.v1.CustomResourceSubresources Old Versions: [apiextensions.v1beta1.CustomResourceSubresources,]
----------------------------------
Definitions with Operations Missing from Toc (Excluding old version):
[storage.v1.CSIDriver]
	 [createStorageV1CSIDriver]
	 [patchStorageV1CSIDriver]
	 [replaceStorageV1CSIDriver]
	 [deleteStorageV1CSIDriver]
	 [deleteStorageV1CollectionCSIDriver]
	 [readStorageV1CSIDriver]
	 [listStorageV1CSIDriver]
	 [watchStorageV1CSIDriver]
	 [watchStorageV1CSIDriverList]
[networking.v1beta1.IngressClass]
	 [createNetworkingV1beta1IngressClass]
	 [patchNetworkingV1beta1IngressClass]
	 [replaceNetworkingV1beta1IngressClass]
	 [deleteNetworkingV1beta1IngressClass]
	 [deleteNetworkingV1beta1CollectionIngressClass]
	 [readNetworkingV1beta1IngressClass]
	 [listNetworkingV1beta1IngressClass]
	 [watchNetworkingV1beta1IngressClass]
	 [watchNetworkingV1beta1IngressClassList]
```

**Comment by tengqm** [comment]
So ... the number of newly promoted resource types is far lower than that of the deprecated ones. Good thing to know.

**Comment by kbhawkey** [comment]
@tengqm , Are you around to approve? I'd like to use the "update" script to test the generation of the component tool pages and kubectl. The API ref will need the updated spec.

**Pull request merged**